app/src/main/java/org/fossasia/openevent/app/data/AbstractObservable.java

```diff
@@ -35,6 +39,18 @@ public AbstractObservableBuilder(ConnectionStatus connectionStatus) {
             return this;
         }
 
+        public AbstractObservableBuilder<T> rateLimiter(String rateLimiterKey) {
+            this.rateLimiterKey = rateLimiterKey;
+
+            return this;
+        }
+
+        public AbstractObservableBuilder<T> withAbstractObservable(AbstractObservable abstractObservable) {
+            this.abstractObservable = abstractObservable;
+
+            return this;
+        }
```

Comment:
Any suggestions on how I could accomplish it without asking in the constructor ? 
Making the rate limiter static in AbstractObservable ?

``` Java
private static final RateLimiter<String> rateLimiter = new RateLimiter<>(Duration.ofMinutes(10));
```

and then using inside the AbstractObservableBuilder.

Reply:
I am adding it in the AbstractObservable only.
I assume you meant `AbstractObservableBuilder` like this:
``` Java
 public static final class AbstractObservableBuilder<T> {
        ...
        private Observable<T> networkObservable;
        private final RateLimiter<String> rateLimiter = new RateLimiter<>(Duration.ofMinutes(10));
```

This won't work since every time we request the repository for data, it creates a new instance of `AbstractObservableBuilder`.
Even if I make the request after 1 sec, it will build a new AbstractObservable and hence a  new rate limiter. The previous one will be gone.

Reply:
Sorry, it seems there has been a mistake in reviewing this. I'll re-review soon